src/config/style_edition.rs

```diff
@@ -0,0 +1,69 @@
+use crate::config::StyleEdition;
+
+/// Defines the default value for the given style edition
+pub(crate) trait StyleEditionDefault {
+    type ConfigType;
+    fn style_edition_default(style_edition: StyleEdition) -> Self::ConfigType;
+}
+
+/// macro to help implement `StyleEditionDefault` for config options
+#[macro_export]
+macro_rules! style_edition_default {
+    ($ty:ident, $config_ty:ty, $default:expr) => {
+        impl crate::config::style_edition::StyleEditionDefault for $ty {
```

Comment:
This is very minor but in case we want to have multiple crates changing all the `crate::*` to `$crate:*` would save some time when moving it later.

```suggestion
        impl $crate::config::style_edition::StyleEditionDefault for $ty {
```